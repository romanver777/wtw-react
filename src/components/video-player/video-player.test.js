/* eslint-disable @typescript-eslint/no-unsafe-call */
/* eslint-disable @typescript-eslint/no-unsafe-member-access */
/* eslint-disable @typescript-eslint/no-unsafe-assignment */
import React from "react";
import TestRenderer from "react-test-renderer";
import VideoPlayer from "./video-player";

describe("VideoPlayer component", () => {
  const data = {
    items: [
      {
        url: "https://www.youtube.com/watch?v=mccs8Ql8m8o",
        name: "Французский трейлер",
        site: "YOUTUBE",
      },
      {
        url: "https://widgets.kinopoisk.ru/discovery/trailer/13420?onlyPlayer=1&autoplay=1&cover=1",
        name: "Фрагмент (содержит спойлер!) (дублированный)",
        site: "KINOPOISK_WIDGET",
      },
      {
        url: "https://www.youtube.com/v/NRT11KuW6L0",
        name: "Немецкий трейлер",
        site: "YOUTUBE",
      },
    ],
    total: 8,
  };
  const isFullScreen = false;
  const props = {
    data,
    isFullScreen,
  };
  const dataEmpty = {};
  const propsEmpty = {
    dataEmpty,
    isFullScreen,
  };

  it("should renders correct with full data", () => {
    const TRenderer = TestRenderer.create(<VideoPlayer {...props} />);
    expect(TRenderer.toJSON()).toMatchSnapshot();
  });
  it("should renders correct with empty data", () => {
    const TRenderer = TestRenderer.create(<VideoPlayer {...propsEmpty} />);
    expect(TRenderer.toJSON()).toMatchSnapshot();
  });
});
