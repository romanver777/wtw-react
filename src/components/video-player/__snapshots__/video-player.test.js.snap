// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayer component should renders correct with empty data 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "background": "rgba(23,2,2,0.9)",
      "display": "flex",
      "fontSize": "11px",
      "height": "100%",
      "justifyContent": "center",
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 4,
    }
  }
>
  Видео не доступно..
</div>
`;

exports[`VideoPlayer component should renders correct with full data 1`] = `
<iframe
  allowFullScreen={false}
  frameBorder="0"
  src="https://youtube.com/embed/mccs8Ql8m8o?autoplay=1&mute=1&controls=0"
  style={
    Object {
      "height": "100%",
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 4,
    }
  }
/>
`;
